include/openthread/link_raw.h

```diff
@@ -351,6 +351,62 @@ otError otLinkRawSetMacKey(otInstance *   aInstance,
                            const uint8_t *aCurrKey,
                            const uint8_t *aNextKey);
 
+/**
+ * This function pointer indication callback to store MAC frame counter to NVM.
+ *
+ * @param[in]  aInstance            A pointer to an OpenThread instance.
+ *
+ */
+typedef void (*otLinkRawMacFcStore)(otInstance *aInstance);
+
+/**
+ * Gets the current MAC frame counter value.
+ *
+ * @param[in]  aInstance  A pointer to an OpenThread instance.
+ *
+ * @returns The MAC frame counter.
+ *
+ */
+uint32_t otLinkRawGetMacFrameCounter(otInstance *aInstance);
```

Comment:
Yes, this is needed. `Mle::Store()` will be called not only because `MacFrameCounter` has reaches the threshold, but could be other reason. For example, when state changed, it will be called to save all the network info to NVM. So this API is needed to get the latest counter in RCP.

Another function is `Mle::AppendLinkFrameCounter`. This will fill the Link Frame Counter TLV in several MLE frames. This requires to get the latest counter in use, too.

Reply:
Can this API guarantee getting the latest frame counter? What if an enh-ack is transmitted just after the API returned?

Reply:
Not guaranteed. But I think it is OK. The differences should be small and host will save (current value + 1000) to NVM. I think it is big enough to cover it.

Reply:
I think we can include the frame counter used in `mRxInfo` in `otRadioFrame`, just like `mAckedWithFramePending`. Thus we can save a API, and the mechanism to notify host. Thoughts?

Reply:
Yes, that could be another way to handle this. Frame counter should be increased by

1. Transmitted secured frame.
2. Transmitted Enh-ACK in response to a received frame.

So we can put the frame counter in TxInfo(from transmitted frame) and RxInfo(for transmitted enh-ack) so that RCP does not need to track the stored value. Host will read each frame info and save it to NVM when necessary.